src/index.ts

```diff
@@ -1,168 +1,27 @@
-  return event;
-};
-
-const handleMarkEvent = (event: MarkEvent): MarkEvent => {
-  console.log(event.ph);
-  return event;
-};
-
-const handleLinkedIDEvent = (event: LinkedIDEvent): LinkedIDEvent => {
-  console.log(event.ph);
-  return event;
-};
-
-events.forEach((event: Event): void => {
-  switch (event.ph) {
-    case EventsPhase.METADATA_EVENTS:
-      handleMetadataEvent(event);
-      break;
-
-    case EventsPhase.DURATION_EVENTS_BEGIN:
-    case EventsPhase.DURATION_EVENTS_END:
-      handleDurationEvent(event);
-      break;
-
-    case EventsPhase.COMPLETE_EVENTS:
-      handleCompleteEvent(event);
-      break;
-
-    case EventsPhase.SAMPLE_EVENTS:
-      handleSampleEvent(event);
-      break;
-
-    case EventsPhase.OBJECT_EVENTS_CREATED:
-    case EventsPhase.OBJECT_EVENTS_SNAPSHOT:
-    case EventsPhase.OBJECT_EVENTS_DESTROYED:
-      handleObjectEvent(event);
-      break;
-
-    case EventsPhase.CLOCK_SYNC_EVENTS:
-      handleClockSyncEvent(event);
-      break;
-
-    case EventsPhase.ASYNC_EVENTS_NESTABLE_START:
-    case EventsPhase.ASYNC_EVENTS_NESTABLE_INSTANT:
-    case EventsPhase.ASYNC_EVENTS_NESTABLE_END:
-      handleAsyncEvent(event);
-      break;
-
-    case EventsPhase.CONTEXT_EVENTS_ENTER:
-    case EventsPhase.CONTEXT_EVENTS_LEAVE:
-      handleContextEvent(event);
-      break;
-    case EventsPhase.INSTANT_EVENTS:
-      handleInstantEvent(event);
-      break;
-
-    case EventsPhase.COUNTER_EVENTS:
-      handleCounterEvent(event);
-      break;
-
-    case EventsPhase.FLOW_EVENTS_START:
-    case EventsPhase.FLOW_EVENTS_STEP:
-    case EventsPhase.FLOW_EVENTS_END:
-      handleFlowEvent(event);
-      break;
-
-    case EventsPhase.MEMORY_DUMP_EVENTS_GLOBAL:
-    case EventsPhase.MEMORY_DUMP_EVENTS_PROCESS:
-      handleMemoryDumpEvent(event);
-      break;
-
-    case EventsPhase.MARK_EVENTS:
-      handleMarkEvent(event);
-      break;
-
-    case EventsPhase.LINKED_ID_EVENTS:
-      handleLinkedIDEvent(event);
-      break;
-    default: {
-      // This case should only be hit if we've forgotten to handle
-      // a type of event, or if the incoming data does not match any
-      // of the expected phases at runtime
-      throw new Error(
-        `Unknown event type "${
-          (event as any).ph
-        }" in trace event ${JSON.stringify(event)}`
-      );
-    }
-  }
-});
+  HermesCPUProfile,
+  HermesSample,
+} from './types/hermesProfileInterfaces';
+import {
+  processSample,
+  constructProcessGraph,
+} from './utils/cpuProfileParsing';
+import { SharedEventProperties } from './types/chromeEventInterfaces';
+
+const HERMES_PROFILE_PATH: string = 'hermes_sample.cpuprofile';
+
+const { samples, stackFrames }: HermesCPUProfile = readData(
+  HERMES_PROFILE_PATH
+);
+const processed_samples: HermesSample[] = processSample(samples, stackFrames);
+const events: SharedEventProperties[] = constructProcessGraph(
+  processed_samples,
+  stackFrames
+);
+// events.push(...traceEvents);
```

Comment:
Ideally we could have just added the traceEvents to the same array, however this gave erroneous results. Not sure why @jevakallio